### PR TITLE
176 Added footer

### PR DESCRIPTION
**Pull request by lucascumsille**
Fixes part of #176
Added footer and some missing styling rules.

Screenshot 2022-01-13 at 13 55 00

### CODE REVIEW

File: scoring/templates/scoring/base.html [hunk 1]
```diff
@@ -47,6 +47,52 @@
         <a class="btn donation ml-2" href="https://www.paypal.com/donate?token=JVAijB-upEdRZMmUvUG04rqaFjj4OfIaEnzewHJZgHGrCCuqOwKlJqaefZttGHMEEo7A45BVdLDBqVkn">Donate</a>
     </nav>
 {% block content %}{% endblock %}
-
+    <footer class="container-fluid bg-dark-grey">
```
Comment (zarino) [inline scoring/templates/scoring/base.html:50]: I wonder whether all of the footer content (ie: everything except the dark background) should go inside a fixed-width container? Just feels like, on a screen like this, the "Council Climate League" logo is _very close_ to the left edge of the window:

Screenshot 2022-01-13 at 15-38-22 SCORING

Up to you though!

Comment (zarino) [reply scoring/templates/scoring/base.html:50]: I think it’s made worse by the `max-width` on `.newsletter-wrapper`, which results in it having lots of space to its right side, so the whole footer feels off-centre on wide screens.

File: scoring/templates/scoring/base.html [hunk 2]
```diff
@@ -47,6 +47,52 @@
...
+                            <input class="name" type="text" id="fname" name="firstname" placeholder="Name">
+                            <input class="last-name" type="email" id="email" placeholder="Email" pattern=".+@globex\.com" size="30" required>
+                            <input class="btn btn-green is--light" type="submit" value="Join us">
+                        </form>
+                    </div>
+                    <div class="media-icons-wrapper mt-3">
+                        <a class="media-icons twitter" href="https://twitter.com/ClimateEmergUK?s=20"></a>
+                        <a class="media-icons facebook" href="https://www.facebook.com/climateemergency/"></a>
+                        <a class="media-icons instagram" href="https://www.instagram.com/climateemergencyuk/"></a>
+                        <a class="media-icons github" href="https://github.com/mysociety/caps.git"></a>
```
Comment (zarino) [inline scoring/templates/scoring/base.html:84]: Don‘t need the `.git` on this line.

File: scoring/templates/scoring/base.html [hunk 3]
```diff
@@ -47,6 +47,52 @@
...
+                        <h5>Receive the lastes news from Climate Emergency UK</h5>
+                        <p class="newsletter-subtitle">Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.</p>
+                        <form class="newsletter-form mt-4">
+                            <input class="name" type="text" id="fname" name="firstname" placeholder="Name">
+                            <input class="last-name" type="email" id="email" placeholder="Email" pattern=".+@globex\.com" size="30" required>
+                            <input class="btn btn-green is--light" type="submit" value="Join us">
+                        </form>
+                    </div>
+                    <div class="media-icons-wrapper mt-3">
+                        <a class="media-icons twitter" href="https://twitter.com/ClimateEmergUK?s=20"></a>
```
Comment (zarino) [inline scoring/templates/scoring/base.html:81]: Don’t need the `?s=20` on this line.

File: scoring/templates/scoring/base.html [hunk 4]
```diff
@@ -47,6 +47,52 @@
...
+                    <div class="newsletter-wrapper">
+                        <h5>Receive the lastes news from Climate Emergency UK</h5>
+                        <p class="newsletter-subtitle">Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.</p>
+                        <form class="newsletter-form mt-4">
+                            <input class="name" type="text" id="fname" name="firstname" placeholder="Name">
+                            <input class="last-name" type="email" id="email" placeholder="Email" pattern=".+@globex\.com" size="30" required>
+                            <input class="btn btn-green is--light" type="submit" value="Join us">
+                        </form>
+                    </div>
+                    <div class="media-icons-wrapper mt-3">
```
Comment (zarino) [inline scoring/templates/scoring/base.html:80]: The links inside this element are missing text content. Look at the `.mysoc-footer__badge` classes in the CAPE footer for an example of how we usually do this, with `padding` and `overflow: hidden` cropping the text labels out of view, but keeping them accessible to assistive devices.

File: scoring/templates/scoring/base.html [hunk 5]
```diff
@@ -47,6 +47,52 @@
...
+                        <a class="media-icons twitter" href="https://twitter.com/ClimateEmergUK?s=20"></a>
+                        <a class="media-icons facebook" href="https://www.facebook.com/climateemergency/"></a>
+                        <a class="media-icons instagram" href="https://www.instagram.com/climateemergencyuk/"></a>
+                        <a class="media-icons github" href="https://github.com/mysociety/caps.git"></a>
+                    </div>
+                    <div class="d-flex d-md-none flex-row align-items-center mt-5">
```
Comment (zarino) [inline scoring/templates/scoring/base.html:86]: Again, the two logo links inside this element need to have text content.

File: scoring/templates/scoring/base.html [hunk 6]
```diff
@@ -47,6 +47,52 @@
...
+                <a class="footer-link" href="#">Contact</a>
+            </div>
+            <div class="col-12 col-md-4">
+                <div>
+                    <div class="newsletter-wrapper">
+                        <h5>Receive the lastes news from Climate Emergency UK</h5>
+                        <p class="newsletter-subtitle">Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.</p>
+                        <form class="newsletter-form mt-4">
+                            <input class="name" type="text" id="fname" name="firstname" placeholder="Name">
+                            <input class="last-name" type="email" id="email" placeholder="Email" pattern=".+@globex\.com" size="30" required>
```
Comment (zarino) [inline scoring/templates/scoring/base.html:76]: You don’t need a `pattern` attribute here. (Even worse, the current one only allows submission of email addresses that end in "@globex.com"!)

File: scoring/templates/scoring/base.html [hunk 7]
```diff
@@ -47,6 +47,52 @@
...
+                <div class="d-none d-md-flex  flex-row align-items-center mt-4 mt-lg-5">
+                    <a class="logo ceuk-logo" href="https://www.climateemergency.uk/"></a>
+                    <div class="d-flex flex-column align-items-start">
+                        <span>Built by</span>
+                        <a class="logo mysociety-logo" href="https://www.mysociety.org/"></a>
+                    </div>
+                </div>
+            </div>
+            <div class="col-12 col-md-4 mb-5 mb-md-0">
+                <a class="footer-link" href="#">Scorecards</a>
```
Comment (zarino) [inline scoring/templates/scoring/base.html:63]: I’m guessing you’ll want `href="{% url 'home' %}"` here.

File: scoring/templates/scoring/base.html [hunk 8]
```diff
@@ -47,6 +47,52 @@
         <a class="btn donation ml-2" href="https://www.paypal.com/donate?token=JVAijB-upEdRZMmUvUG04rqaFjj4OfIaEnzewHJZgHGrCCuqOwKlJqaefZttGHMEEo7A45BVdLDBqVkn">Donate</a>
     </nav>
 {% block content %}{% endblock %}
-
+    <footer class="container-fluid bg-dark-grey">
+        <div class="row">
+            <div class="col-12 col-md-4 d-flex flex-column justify-content-between mb-5 mb-md-0">
+                <div class="scorecards-logo"></div>
+                <div class="d-none d-md-flex  flex-row align-items-center mt-4 mt-lg-5">
```
Comment (zarino) [inline scoring/templates/scoring/base.html:54]: As mentioned below, the links inside this element are missing text content.

**Review by zarino — CHANGES_REQUESTED** [review]
Styling looks good. Couple of comments on the markup though.

File: scoring/templates/scoring/base.html (hunk 6 above)
Comment (lucascumsille) [reply scoring/templates/scoring/base.html:76]: To be fair this last one is just for markup and styling purposes, I'll ask CEUK how they manage newsletters, mailchimp or any other platform they use. Then I'll put the correct code.

File: scoring/templates/scoring/base.html [hunk 9]
```diff
@@ -47,6 +47,54 @@
...
+                            </form>
+                        </div>
+                        <div class="media-icons-wrapper mt-4">
+                            <a class="scorecard-footer__badge scorecard-footer__badge--twitter" href="https://twitter.com/ClimateEmergUK">Twitter</a>
+                            <a class="scorecard-footer__badge scorecard-footer__badge--facebook" href="https://www.facebook.com/climateemergency/">Facebook</a>
+                            <a class="scorecard-footer__badge scorecard-footer__badge--instagram" href="https://www.instagram.com/climateemergencyuk/">Instagram</a>
+                            <a class="scorecard-footer__badge scorecard-footer__badge--github" href="https://github.com/mysociety/caps">Github</a>
+                        </div>
+                        <div class="d-flex d-md-none flex-row align-items-center mt-5">
+                            <a class="logo ceuk-logo" href="https://www.climateemergency.uk/"></a>
```
Comment (zarino) [inline scoring/templates/scoring/base.html:88]: Still missing text inside this link.

File: scoring/templates/scoring/base.html [hunk 10]
```diff
@@ -47,6 +47,54 @@
...
+                            <div class="d-flex flex-column align-items-start">
+                                <span>Built by</span>
+                                <a class="logo mysociety-logo" href="https://www.mysociety.org/"></a>
```
Comment (zarino) [inline scoring/templates/scoring/base.html:91]: Still missing text inside this link.

**Review by zarino — CHANGES_REQUESTED** [review]

**Review by zarino — APPROVED** [review]

**Pull request merged**